### DISCUSSION TITLE
How to get all fields from inside a model_serializer

### DISCUSSION THREAD

**Discussion by shadycuz** [pydantic/pydantic#8193] [General]
I have a use case where at the end of pydantic processing I need to walk all fields and nested fields looking for specific criteria. I thought I could use a `model_serializer` function to do that. 

```python
@model_serializer()
def resolve_variables(self) -> Dict[str, Any]:
     """Resolves variables in the configuration file."""
     data = self.model_dump(exclude_none=True)
     return resolve_values(data, data)
```
But when I call this code I get the following error:
> pydantic_core._pydantic_core.PydanticSerializationError: Error calling function `resolve_variables`: RecursionError

It's from using `model_dump` from inside the `model_serializer`. Is there a similar method that can be called to get a dictionary like structure of all the model fields and values?

**Comment by azmeuk** [comment]
I am not sure how to achieve this either. Having a `value` attribute like in `field_serializer` would be a bliss in that situation.
